internal/pkg/term/progress/cloudformation.go

```diff
@@ -85,6 +124,90 @@ func (c *regularResourceComponent) Done() <-chan struct{} {
 	return c.done
 }
 
+// ecsServiceResourceComponent can display an ECS service created with CloudFormation.
+type ecsServiceResourceComponent struct {
+	// Required inputs.
+	cfnStream    <-chan stream.StackEvent   // Subscribed stream to initialize the deploymentRenderer.
+	ecsDescriber stream.ECSServiceDescriber // Client needed to create an ECSDeploymentStreamer.
+	logicalID    string                     // LogicalID for the service.
+
+	// Optional inputs.
+	group      *errgroup.Group // Existing group to catch ECSDeploymentStreamer errors.
+	ctx        context.Context // Context for the ECSDeploymentStreamer.
+	renderOpts RenderOptions
+
+	// Sub-components.
+	resourceRenderer   Renderer
+	deploymentRenderer Renderer
+
+	done chan struct{}
+	mu   sync.Mutex
+}
+
+// Listen creates a deploymentRenderer if the service is being created, or updated.
+func (c *ecsServiceResourceComponent) Listen() {
+	var prevStatus string
+	for ev := range c.cfnStream {
+		if c.logicalID != ev.LogicalResourceID {
+			continue
+		}
+		curStatus := ev.ResourceStatus
+		if curStatus != prevStatus && cloudformation.StackStatus(curStatus).UpsertInProgress() {
+			c.mu.Lock()
+
+			// Start a deployment renderer if a service deployment is happening.
+			arn := ecs.ServiceArn(ev.PhysicalResourceID)
+			cluster, _ := arn.ClusterName() // Errors can't happen on valid ARNs.
+			service, _ := arn.ServiceName()
```

Comment:
How about service names?

Reply:
Same! I'll update the comment in a followup PR.